Rebase on Presto 306

BenoitHanotte opened the pull request:
rebase on 306
+ use MetastoreLocator API
+ Throttle Hive split discovery

zhenxiao committed:
Add max data size per query

Signed-off-by: Raphael CLAUDE <r.claude@criteo.com>

piyushnarang committed:
Fix issue: 10099 - create table failed when using viewfs

Signed-off-by: Raphael CLAUDE <r.claude@criteo.com>

piyushnarang committed:
Criteo light travis yaml

Purpose is to avoid timeout & waisting cpu cycles on unused/unmodified
features
* Comment out some database connectors modules
* Comment out SqlServer docker tests
* Turn off travis slack updates

piyushnarang commented:
This change link highlights the patches we have cherry-picked, our custom internal patches and any workarounds - https://github.com/criteo-forks/presto/compare/master...piyushnarang:206-merge

piyushnarang commented:
cc @szehonCriteo

szehonCriteo commented:
+1, it looks good to me

Commit:
Throttle Hive splits discovery from the HiveSplitSource

Add `hive.max-splits-per-sec` configuration option to allow throttling
the number of splits released by the HiveSplitSource to provide strong
guarantee on the rate of read operations on the namenode.

Use Guava's RateLimiter to simplify code

Set hive.max-splits-per-second to Integer.MAX_VALUE by default

qinghui-xu committed:
Query history extension

* PR(origin) of query history extension

This is the original PR for creating query history extension.

The architecture is:
(Read part)
________                                    __________
                     ______________
|        |--- request for a query's info -->|          |---- QueryId
----->|              |
| web ui |                                  | REST api |
                   | QueryManager |
|________|<-- Serialized QueryInfo (json) --|__________|<---- QueryInfo
---|______________|

^              |

|              |

QueryInfo       QueryId

|              |

|              |
____                   ______________
                                     |______________v
|    |----- Json ----->|              |-- Deserialize json to QueryInfo
-->|              |
| DB |                 | HistoryStore |
                                    | QueryTracker |
|____|<-- SQL Query ---|______________|<------- request for QueryId
-------|______________|

(write part)
___________________                       ______________
                                 ____
|   QueryTracker    |                     |              |
                               |    |
| (event when query |-- Save QueryInfo -->| HistoryStore |--Serialize
QueryInfo (json) ->| DB |
| finished)         |                     |              |
                               |    |
|___________________|                     |______________|
                               |____|

* Plug QueryHistoryStore directly to REST API

So that we can directly send json string back to client web ui, instead
of having to reparse the string into some complex java object which is
error prone.
Also, this should save some cpu cycles by removing the json
deserialization/serialization overhead.

By doing this, we also remove some customized json serialization logic.

* Add index to the query history table

Remove the test with h2 because the create table with index syntax is
unsupported.

* Keep the QueryTracker signature

To be easier for future upstream rebase.

* QueryHistoryStore architecture

* [Checkstyle] Fix an error

* Put the query-history-extension.properties under etc dir

Commit:
Discover meastore URIs using consul

Add the DisoveryMetastoreLocator to resolve the metastore uri from
consul.
